tests/foreman/api/test_rhcloud_inventory.py

```diff
@@ -78,24 +80,41 @@ def test_rhcloud_inventory_api_e2e(
         5. JSON files inside report can be parsed
         6. metadata.json lists all and only slice JSON files in tar
         7. Host counts in metadata matches host counts in slices
-        8. Assert Hostnames, IP addresses, and installed packages are present in report.
+        8. metadata contains source and foreman_rh_cloud_version keys.
+        9. Assert Hostnames, IP addresses, infrastructure_type, and installed packages
+            are present in report.
 
     :CaseImportance: Critical
 
-    :BZ: 1807829, 1926100, 1965234
+    :BZ: 1807829, 1926100, 1965234, 1824183, 1879453
     """
     org, ak = organization_ak_setup
     virtual_host, baremetal_host = rhcloud_registered_hosts
     local_report_path = robottelo_tmp_dir.joinpath(f'{gen_alphanumeric()}_{org.id}.tar.xz')
     # Generate report
-    generate_inventory_report(rhcloud_sat_host, org)
```

Comment:
@adarshdubey-star No need to remove `generate_inventory_report(rhcloud_sat_host, org)`, it does the same thing for which you have added new code.